driver-sync/src/main/com/mongodb/client/MongoCluster.java

```diff
@@ -352,4 +357,83 @@ public interface MongoCluster {
+     * and there is at least one of the following pieces of information to report:
+     * {@link ClientBulkWriteException#getWriteConcernErrors()}, {@link ClientBulkWriteException#getWriteErrors()},
+     * {@link ClientBulkWriteException#getPartialResult()}.
+     * @throws MongoException Only if the operation is unsuccessful.
+     * @since 5.3
+     * @mongodb.server.release 8.0
+     * @mongodb.driver.manual reference/command/bulkWrite/ bulkWrite
+     */
+    ClientBulkWriteResult bulkWrite(
+            List<? extends ClientWriteModel> models,
+            ClientBulkWriteOptions options) throws ClientBulkWriteException;
+
+    /**
+     * Executes a client-level bulk write operation.
+     * This method is functionally equivalent to {@link #bulkWrite(ClientSession, List, ClientBulkWriteOptions)}
+     * with the {@linkplain ClientBulkWriteOptions#clientBulkWriteOptions() default options}.
+     *
+     * @param clientSession The {@linkplain ClientSession client session} with which to associate this operation.
+     * @param models The {@linkplain ClientWriteModel individual write operations}.
+     * @return The {@link ClientBulkWriteResult} if the operation is successful.
+     * @throws ClientBulkWriteException If and only if the operation is unsuccessful or partially unsuccessful,
+     * and there is at least one of the following pieces of information to report:
+     * {@link ClientBulkWriteException#getWriteConcernErrors()}, {@link ClientBulkWriteException#getWriteErrors()},
+     * {@link ClientBulkWriteException#getPartialResult()}.
+     * @throws MongoException Only if the operation is unsuccessful.
+     * @since 5.3
+     * @mongodb.server.release 8.0
+     * @mongodb.driver.manual reference/command/bulkWrite/ bulkWrite
+     */
+    ClientBulkWriteResult bulkWrite(
+            ClientSession clientSession,
+            List<? extends ClientWriteModel> models) throws ClientBulkWriteException;
+
+    /**
+     * Executes a client-level bulk write operation.
+     *
+     * @param clientSession The {@linkplain ClientSession client session} with which to associate this operation.
+     * @param models The {@linkplain ClientWriteModel individual write operations}.
+     * @param options The options.
+     * @return The {@link ClientBulkWriteResult} if the operation is successful.
+     * @throws ClientBulkWriteException If and only if the operation is unsuccessful or partially unsuccessful,
+     * and there is at least one of the following pieces of information to report:
+     * {@link ClientBulkWriteException#getWriteConcernErrors()}, {@link ClientBulkWriteException#getWriteErrors()},
+     * {@link ClientBulkWriteException#getPartialResult()}.
+     * @throws MongoException Only if the operation is unsuccessful.
+     * @since 5.3
+     * @mongodb.server.release 8.0
+     * @mongodb.driver.manual reference/command/bulkWrite/ bulkWrite
+     */
+    ClientBulkWriteResult bulkWrite(
```

Comment:
`BULK-TODO` Document that retryability is supported and the conditions for which retryability is determined (the [spec requires this](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.md#:~:text=Drivers%20MUST%20document%20operations%20that%20support%20retryable%20behavior%20and%20the%20conditions%20for%20which%20retryability%20is%20determined)).